Rework getting started pages (#273)
neo4j/docs-graphql: modules/ROOT/pages/index.adoc

If you are new to Neo4j and GraphQL take a look at xref:getting-started/index.adoc[Creating a new project] and xref:getting-started/toolbox.adoc[Neo4j GraphQL Toolbox] to learn the fundamentals of the Neo4j GraphQL Library and how to create GraphQL APIs backed by a Neo4j graph database.

== How it works

The Neo4j GraphQL Library requires a set of type definitions that describes the shape of your graph data.
It can generate an entire executable schema with all of the additional types needed to execute queries and mutations to interact with your Neo4j database.

For every query and mutation that is executed against this generated schema, the Neo4j GraphQL Library generates a single Cypher query which is executed against the database.
This eliminates the https://www.google.com/search?q=graphql+n%2B1[N+1 Problem], which can make GraphQL implementations slow and inefficient.

> The Neo4j GraphQL Library features:

Reviewer: Sorry, I'm not able to comment on the section I want here 😬 

This is just a minor point but in the features list filtering, sorting and pagination all belong in the same bullet point

== Interaction

// Modify link below and add info about aura graphql

> In xref::getting-started/graphql-aura.adoc[] as well as xref::getting-started/graphql-self-hosted.adoc[], Apollo Server is used to host the GraphQL schema, so you can interact directly with your API with no frontend.

Reviewer: We don't actually use yoga instead of Apollo Server in Aura. We also don't really want to be opinionated about which server self-hosted customers use as the library should work with any graphql server

```suggestion
In xref::getting-started/graphql-aura.adoc[] as well as xref::getting-started/graphql-self-hosted.adoc[], a GraphQL server is used to serve the GraphQL schema, so you can interact directly with your API with no frontend.
```

In case you prefer to use frontend frameworks, these are some clients that interact with GraphQL APIs:

- https://reactjs.org/[React] - support through https://www.apollographql.com/docs/react/[Apollo Client]
- https://vuejs.org/[Vue.js] - support through https://apollo.vuejs.org/[Vue Apollo]
- https://angularjs.org/[AngularJS] - support through https://apollo-angular.com/docs/[Apollo Angular].

== Deployment

> There are a variety of methods for deploying GraphQL APIs.

Reviewer: It might be good to mention something here about using Aura to deploy your API as that would be our preference

`NUMBER` in the suffix is simply an incrementing release number in each phase.

== Requirements

. https://neo4j.com/product/auradb/[Neo4j AuraDB] version 5.x with APOC core plugin or https://neo4j.com/deployment-center/#gdb-selfmanaged[Neo4j Database].

> Note that version 5.15 or higher is required to use the xref:/directives/indexes-and-constraints.adoc#_vector_index_search[`@vector` directive].

Reviewer: We were actually discussing last week changing this wording as it's inaccurate. We require users to be on the latest patch of a major version. There's also not much reason they shouldn't be keeping up to date

Also Aura always ships with APOC and there is no 5.x dbs in Aura anymore

```suggestion
. https://neo4j.com/product/auradb/[Neo4j AuraDB] version 2025.x or https://neo4j.com/deployment-center/#gdb-selfmanaged[Neo4j Database] running the latest 5.x version or above with the APOC core plugin.
```

> . https://nodejs.org/en/[Node.js] 20+.

Reviewer: I think this only belongs in the self hosted specific docs as this is not important for people deploying in Aura

```suggestion
```
